### ISSUE TITLE
Plugin not working in new project with latest Ionic / Angular Version

### ISSUE THREAD

**Issue by nu3trition** [sneas/ionic-native-http-connection-backend#32]
Hi there,

thanks for your work - this plugin is exactly what I was looking for. Unfortunately I can't get it to work. I created a blank project with the environment listed below and followed the instructions.

You can find my code here: https://github.com/nu3trition/cors-buster

This is my ionic info:

cli packages: (/usr/local/lib/node_modules)

    @ionic/cli-utils  : 1.19.2
    ionic (Ionic CLI) : 3.20.0

global packages:

    cordova (Cordova CLI) : 8.0.0 

local packages:

    @ionic/app-scripts : 3.1.8
    Cordova Platforms  : ios 4.5.4
    Ionic Framework    : ionic-angular 3.9.2

System:

    ios-deploy : 1.9.2 
    Node       : v6.11.4
    npm        : 5.6.0 
    OS         : macOS Sierra
    Xcode      : Xcode 9.1 Build version 9B55 

What am I doing wrong?

**Comment by sneas** [comment]
Hi @nu3trition ,

`this.http.get('https://www.dropbox.com', {responseType: "arraybuffer"})` -----> `this.http.get('https://www.dropbox.com', {responseType: "text"})` will help 😉

**Comment by nu3trition** [comment]
Hi sneas,

thanks for the fast response!

Indeed it works when I switch to responseType "text".
Unfortunately I'm forced to use ArraBuffer as responseType, as the API (not dropbox.com, that's only a placeholder without CORS Header) sends me ISO 8859-1 without telling me the format in the response header. That's why I have to convert it to UTF-8 via a Uint8Array.

Is there a way to use this promising plugin with ArrayBuffer? That would be awesome...

**Comment by sneas** [comment]
Hi @nu3trition, with your help we can update the lib's code to support ArrayBuffer responses.

Can you provide the sequence of actions (`convertResponseDataToArrayBuffer()` function's  body) to be performed to convert nativeHttp `response.data` to `ArrayBuffer`? So the function could be used with the example below:

```
this.platform.ready().then(() => {
    this.nativeHttp.get('array-buffer-endpoint-url', {a: 'b'}, {})
        .then((response) => {
            const arrayBuffer = convertResponseDataToArrayBuffer(resposne.data);
        });
});
```

Also it will be very helpful to know any public API providing ISO 8859-1 responses.

**Comment by nu3trition** [comment]
Hi @sneas,

thanks alot for your help!

At the moment I'm using Angulars HttpClientModule to properly receive ArrayBuffer responses. This is the code for the API call:

let params = new HttpParams().set('lang', 'de')
params = params.append('query', 'string')

this.httpClientModule.get('http://weirdest-api-ever.de/item.xml', {params: params, responseType: "arraybuffer"})
      .subscribe(res => {
        let sl = new Uint8Array(res)
        let resIso = ""
        sl.forEach((x) => {
          resIso += String.fromCharCode(x)
        })

I tried to get the results via nativeHttp, but I didn't manage to receive a real ArrayBuffer. There is no header telling the client what encoding is being used for the response. It just delivers ISO 8859-1. As there is no way to tell whether the response body is UTF-8 or ISO, the client treats it as UTF-8 when ResponseType is "text". When this happens, there is no way back (at least I can't imagine one) and you get funny characters when the response contains for example German umlauts (Ä,ä,Ö).

I searched the web for an open API delivering ISO 8859-1. I assume most of them died years ago or embraced the future with UTF-8. Probably I exhumed the last API with this funny encoding and no CORS header. Unfortunately I have no control over the server and my app depends on that antique piece of software for the next months.

Hopefully there is a way to get this to work.
Thanks again for your help!
